cyences_app_for_splunk/bin/cyences_inventory_backfill.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import sys
+import time
+
+from splunklib.searchcommands import (
+    dispatch,
+    GeneratingCommand,
+    Configuration,
+    Option,
+    validators,
+)
+
+import splunklib.client as client
+import splunklib.results as results
+import cs_utils
+import logging
+import logger_manager
+
+logger = logger_manager.setup_logging("inventory_backfill", logging.INFO)
+
+
+def handle_results(response):
+    reader = results.JSONResultsReader(response)
+    for result in reader:
+        if isinstance(result, results.Message):
+            logger.warning("{}: {}".format(result.type, result.message))
+
+
+@Configuration()
+class CyencesInventoryBackfill(GeneratingCommand):
+    search_prefix = Option(name="search_prefix", require=False, default="-")
+    earliest_time = Option(name="earliest_time", require=False, default="-24h")
+    latest_time = Option(name="latest_time", require=False, default="now")
+
+    def generate(self):
+        try:
+            self.session_key = cs_utils.GetSessionKey(logger).from_custom_command(self)
+            self.conf_manger = cs_utils.ConfigHandler(logger, self.session_key)
+
+            saved_searches = self.conf_manger.get_saved_searches()
+
+            for saved_search in saved_searches:
+                name = saved_search["name"]
+                if name.startswith(self.search_prefix) and (not cs_utils.is_true(saved_search["content"].get("disabled"))):
+
+                    if name == "Device Inventory - Windows Defender":
+                        SEARCH_QUERY = '| search `cs_windows_defender` EventCode=1151 `cs_windows_defender_max_timerange` | `cs_windows_defender_inventory_fill_search` | where SEARCHNOTHING="SEARCHNOTHING"'
```

Comment:
Why we have this query hard-coded?

Reply:
it's to limit the time range to last 24 hour using this macro:: cs_windows_defender_max_timerange

don't know the actual reason but previously, it is defined like this and rest of the search is also in macro (cs_windows_defender_inventory_fill_search) so that it's easy to use at both places (in actual saved search and in backfill search)